simulations/ContestSimulation.java

```diff
@@ -0,0 +1,192 @@
+            .post("/team/clarifications/add")
+            .formParam("team_clarification[recipient]", "dummy")
+            .formParam("team_clarification[subject]", "#{clarification_subject}")
+            .formParam("team_clarification[message]", "#{user} needs help")
+            .formParam("team_clarification[_token]", "#{csrftoken}")
+            .formParam("submit", "")
+        );
+    }
+
+    /*********************** Spectator Scenario ****************************************************************
+     * Scenario that just grabs the public scoreboard every 30 seconds. The scoreboard page autorefreshes
+     * every 30s, so this is a good approximation of their activity.
+     */
+    HttpRequestActionBuilder publicScoreboard = http("Public Scoreboard").get("/public");
+    ChainBuilder monitorScoreboard = exec(poll().every(30).exec(publicScoreboard)).pause(Duration.ofMinutes(MAX_MINUTES));
+    ScenarioBuilder spectatorScenario = scenario("SpectatorScenario").exec(monitorScoreboard);
+
+
+    /*********************** Team Scenario *********************************************************************
+     * This pretends to be a team, it polls the team page every 30s and randomly performs various actions
+     */
+    // Feeder for team accounts
+    AtomicInteger counter = new AtomicInteger(1);
+    Iterator<Map<String,Object>> feeder =
+        Stream.generate((Supplier<Map<String,Object>>) () -> {
+            String username = String.format("gatling%04d", counter.getAndIncrement());
+            return Collections.singletonMap("user", username);
+        }
+    ).iterator();
+
+    HttpRequestActionBuilder teamPage = http("Team Page").get("/team");
+    ChainBuilder teamSubmitC = submit("c", "test-hello.c");
+    ChainBuilder teamSubmitCpp = submit("cpp", "test-hello.c++");
+    ChainBuilder teamSubmitJava = submit("java", "test-hello.java");
+    ChainBuilder teamSubmitKotlin = submit("kt", "test-hello.kt");
+    ChainBuilder teamSubmitPython = submit("py3", "test-hello.py3");
+    ChainBuilder teamRequestClarification = requestClarification();
+    ChainBuilder teamProblems = exec(http("Team Problemset").get("/team/problems"));
+    ChainBuilder teamScoreboard = exec(http("Team Scoreboard").get("/team/scoreboard"));
+    ScenarioBuilder teamScenario = scenario("TeamScenario")
+        .feed(feeder)
+        .exec(http("Homepage").get("/"))
+        .exec(publicScoreboard) // Fetch the public scoreboard
+        .pause(1,5)               // Brief delay
+        .exec(login())            // Log in as the team
+
+        // Mimic the behavior of the team page, it autorefreshes every 30s
+        .exec(
+            poll().pollerName("teampage")
+            .every(30)
+            .exec(teamPage)
+        )
+
+        // While we're polling the teampage, lets pretend to do some actions like a real team might
+        .repeat(NUM_TEAM_ACTIONS, "i").on(
+            exec(
+                randomSwitch().on(
```

Comment:
These are the random actions that a team will do, and the probabilities of each one.